### PR TITLE
fix: `publish script` to work for workspace members

### LINKED ISSUE #1084: [scripts]: publish release script doesn't support `workspace members`

### PR DESCRIPTION
**Pull request by niklasad1**
Close #1084

### CODE REVIEW

File: scripts/publish.sh [hunk 1]
```diff
@@ -74,24 +75,11 @@ for CRATE_DIR in ${ORDER[@]}; do
 		fi
 	done
 
-	# Wait again to make sure that the new version is published and available.
```
Comment (niklasad1) [inline scripts/publish.sh:77]: Latest `cargo publish`does indeed wait for registry to available so this is not needed anymore.

File: scripts/publish.sh [hunk 2]
```diff
@@ -74,24 +75,11 @@ for CRATE_DIR in ${ORDER[@]}; do
...
-			echo "#### Got $NAME@$REMOTE_VERSION but expected $NAME@$VERSION. Retrying..."
-		fi
-	done
 	cd - > /dev/null
 done
 
 echo "Tagging jsonrpsee@$VERSION"
 set -x
-git tag -a v$VERSION -m "Version $VERSION"
+git tag -a -s v$VERSION -m "Version $VERSION"
```
Comment (niklasad1) [inline scripts/publish.sh:83]: sign the tag

**Review by jsdw — APPROVED** [review]
Looks sane to me if it appears to work when run :)

**Review by lexnv — APPROVED** [review]
Nice! 👍 

I'm guessing the git-lab test is failing because of a difference in rust versions.

The term `deprecated method` has been replaced by `deprecated associated function` for `tests/ui/incorrect/rpc/rpc_deprecated_method.rs`

**Pull request merged**